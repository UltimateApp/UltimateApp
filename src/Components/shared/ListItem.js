import React from 'react';
import { StyleSheet, View } from 'react-native';

import theme from '../../styles/theme.style';

const ListItem = props => {
  return <View style={styles.listItem}>{props.children}</View>;
};

const styles = StyleSheet.create({
  listItem: {
    marginBottom: 10,
    padding: 10,
    backgroundColor: theme.BACKGROUND_COLOR_LIGHT,
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.3,
    shadowRadius: 3,
    elevation: 3,
  },
});

export default ListItem;
