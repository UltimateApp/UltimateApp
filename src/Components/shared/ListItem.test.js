import React from 'react';
import renderer from 'react-test-renderer';
import { Text } from 'react-native';

import ListItem from './ListItem';

describe('<ListItem />', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <ListItem>
          <Text>Hello world</Text>
        </ListItem>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
