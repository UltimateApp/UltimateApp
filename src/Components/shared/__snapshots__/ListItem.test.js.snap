// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "elevation": 3,
      "marginBottom": 10,
      "padding": 10,
      "shadowOffset": Object {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.3,
      "shadowRadius": 3,
    }
  }
>
  <Text>
    Hello world
  </Text>
</View>
`;
